test/e2e/internal/utils/match/status.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package match
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+)
+
+// status represents the expected value of first field in the status log.
+type status = string
+
+// StateKey represents the expected value of second and third fields in status log.
+type StateKey struct {
+	Digest string
+	Name   string
+}
+
+type state struct {
+	uint // padding to avoid zero-based pointer
+}
+
+type edge = struct {
+	from *state
+	to   *state
+}
+
+// stateMachine with edges named after known status.
+type stateMachine struct {
+	edges map[status][]edge
+	start *state
+	end   *state
+}
+
+func newStateMachine(cmd string) (sm *stateMachine) {
```

Comment:
It seems the named parameter is not necessary.
```suggestion
func newStateMachine(cmd string) *stateMachine {
```